cmd/run.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"maps"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/C-L-I-M/chapter-dong-dong/config"
+	"github.com/C-L-I-M/chapter-dong-dong/discord"
+	"github.com/C-L-I-M/chapter-dong-dong/scappers"
+	"github.com/bwmarrin/discordgo"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+func run(cmd *cobra.Command, _ []string) {
+	cfg, err := config.Load()
+	cobra.CheckErr(err)
+
+	session, err := discordgo.New("Bot " + cfg.DiscordToken)
+	cobra.CheckErr(err)
+
+	channels, err := discord.LoadChannels(session, cfg.ServerId, cfg.StaticChannels)
+	cobra.CheckErr(err)
+
+	cobra.CheckErr(channels.ResolveConfigDiff(slices.Sorted(maps.Keys(cfg.Sagas))))
+
+	var wg sync.WaitGroup
+	wg.Add(len(cfg.Sagas))
+
+	for sagaSlug, saga := range cfg.Sagas {
+		scrapper, err := scappers.NewScrapper(saga.SchedulingMode, saga.Parameters)
+		cobra.CheckErr(err)
+
+		go func(saga *config.Saga, scrapper scappers.Scrapper) {
+			defer wg.Done()
+
+			ticker := time.NewTicker(saga.Interval)
+			defer ticker.Stop()
+			for range ticker.C {
+				log.Info(sagaSlug + ": tick start")
+				ctx := scappers.FromSaga(sagaSlug, saga)
+				chapters, err := scrapper.Scrap(ctx)
+				if err != nil {
+					log.Errorf("%s: failed to scrap: %v", sagaSlug, err)
+					continue
+				}
+
+				for _, chapter := range chapters {
```

Comment:
Do you actually want to send one message per new chapter? Wouldn't a more custom message be better?

Reply:
Yes, come do it yourself